### PR TITLE
Update to v0.14.0-rc3

### PR DESCRIPTION
**Pull request by JordanMartinez**
Backlinking to purescript/purescript#3942

### CODE REVIEW

**Comment by JordanMartinez** [comment]
This also seems to be blocked by its test dependency on `quickcheck`

**Comment by JordanMartinez** [comment]
For the role declaration rationale, see [the opening comment of #33](https://github.com/purescript/purescript-ordered-collections/pull/33#issue-485893983)

**Review by thomashoneyman — APPROVED** [review]
Might need a quick review from @hdgarrood or @kl0tl just to verify the role annotation, but otherwise 👍

**Review by hdgarrood — APPROVED** [review]

**Pull request merged**
